src/pip/_internal/req/req_install.py

```diff
@@ -218,7 +215,7 @@ def format_debug(self) -> str:
     def name(self) -> Optional[str]:
         if self.req is None:
             return None
-        return pkg_resources.safe_name(self.req.name)
+        return self.req.name
```

Comment:
This will _probably_ cause some change in logging in some obscure edge cases, but should not have any real functional consequences. `pkg_resources.safe_name()` is pretty useless (it does not correctly normalise `.` and upper cases) and we already call `canonicalize_name` for all relevant places.